clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1125,28 +1135,40 @@ class ArchiveFileHandler final : public FileHandler {
         continue;
       }
 
+      std::unique_ptr<FileHandler> FH{nullptr};
+      std::unique_ptr<MemoryBuffer> Buf{nullptr};
+      StringRef Ext("o");
+      if (BundlerConfig.FilesType == "aocr" ||
+          BundlerConfig.FilesType == "aocx")
+        Ext = BundlerConfig.FilesType;
+
       auto BinOrErr = C.getAsBinary();
       if (!BinOrErr) {
+        /* Not a recognized binary file.  Specifically not an object file */
         if (auto Err = isNotObjectErrorInvalidFileType(BinOrErr.takeError()))
           return Err;
-        continue;
-      }
-
-      auto &Bin = BinOrErr.get();
-      if (!Bin->isObject())
-        continue;
-
-      auto Obj = std::unique_ptr<ObjectFile>(cast<ObjectFile>(Bin.release()));
-      auto Buf = MemoryBuffer::getMemBuffer(Obj->getMemoryBufferRef(), false);
 
-      auto ChildNameOrErr = C.getName();
-      if (!ChildNameOrErr)
-        return ChildNameOrErr.takeError();
+        if (BundlerConfig.FilesType == "aoo") {
+          /* Handle bundled BC Files */
```

Comment:
Should we be using C++ style comments?

Reply:
Changed to C++ style comments to be consistent with other comments in rest of the file.